src/backends/mysql/blob.cpp

```diff
@@ -18,43 +21,99 @@
 using namespace soci;
 using namespace soci::details;
 
-mysql_blob_backend::mysql_blob_backend(mysql_session_backend &session)
-    : session_(session)
+mysql_blob_backend::mysql_blob_backend(mysql_session_backend &)
+    : details::trivial_blob_backend()
 {
-    throw soci_error("BLOBs are not supported.");
 }
 
 mysql_blob_backend::~mysql_blob_backend()
 {
 }
 
-std::size_t mysql_blob_backend::get_len()
+static unsigned char decode_hex_digit(char c)
 {
-    throw soci_error("BLOBs are not supported.");
+    unsigned char i = static_cast<unsigned char>(tolower(c));
+
+    assert((i >= '0' && i <= '9') || (i >= 'a' && i <= 'f'));
```

Comment:
I strongly disagree with using `assert` here. The data comes from some external source and so can be invalid, it shouldn't kill any program using SOCI, i.e. there should be actual error handling.